target/snitch_cluster/Makefile

```diff
@@ -164,6 +165,11 @@ $(GENERATED_DIR)/memories.json: ${CFG} ${CLUSTER_GEN_PREREQ} | $(GENERATED_DIR)
 $(GENERATED_DIR)/bootdata.cc: ${CFG} ${CLUSTER_GEN_PREREQ} | $(GENERATED_DIR)
 	$(CLUSTER_GEN) -c $< -o $(GENERATED_DIR) --bootdata
 
+# REGGEN regfile
+$(PERIPH_DIR)/snitch_cluster_peripheral_reg_top.sv: $(PERIPH_DIR)/snitch_cluster_peripheral_reg.hjson
+	@echo "[REGGEN] Generating $@"
+	$(REGGEN) -r -t $(PERIPH_DIR) $<
+
```

Comment:
We should probably either have a PHONY `rtl` target, which is invoked by `all` and generates this file, together with the cluster wrapper and so on, or we should add them as a dependency of the relevant targets, e.g. `bin/snitch_cluster.vsim` (possibly grouped in an `RTL_SOURCES` variable). Best option would be to have both.

Also `snitch_cluster_peripheral_reg_top.sv` is not the only file generated by this command right? So the others should be listed next to it, before the `:`, as well.

Reply:
I added a `rtl` target. I think the `RTL_SOURCES` is already done now anyway, or not?

Reply:
The command generates both `reg_pkg` and `top_pkg` at the same time, so when you add two targets, it will do it twice. Instead I now added one of them as a prerequisite of the other to avoid this.

Reply:
But something still does not work with the prerequisites somehow in the CI. I will check what is wrong

Reply:
Yeah, sorry I added the explanation before, but eliminated it by mistake. Give me 30mins and I'll write it again

Reply:
The `rtl` target is convenient because you can separately choose to build only the RTL sources. However, it has to be separately invoked before the simulation models can be built, or this would fail cause the files aren't available. This means we need to add it to the CI step, to the tutorial and also to our flow every time those files have to be updated.

In my opinion, it is just much better to have this done transparently by the build system behind the scenes, properly encoding the dependencies in the Makefiles. This would be done by adding the `RTL_SOURCES` as a prerequisite of all targets which depend on them (i.e. those whose rules would fail if the files were missing).

Reply:
Ah yes, you're right about that.

I see the fix, and it's quite tricky. Do you understand why it works after moving this up `COMMON_BENDER_FLAGS += -t snitch_cluster`?

I would've expected that the expansion in the right-hand side of e.g. the `VSIM_SOURCES` assignment would be deferred, and thus expanded after the addition to `COMMON_BENDER_FLAGS` had already been read, so the order wouldn't matter https://www.gnu.org/software/make/manual/html_node/Reading-Makefiles.html.

Reply:
Actually no, according to the manual I would also expect that it does not make a difference

Reply:
I will resolve this, since I reverted the commit. the `snitch_cluster` target is not needed anymore for the peripherals.